userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -761,21 +825,104 @@ uint8_t* sinsp_filter_check_thread::extract(sinsp_evt *evt, OUT uint32_t* len, b
 	case TYPE_ENV:
 		{
 			m_tstr.clear();
-
 			uint32_t j;
 			const auto& env = tinfo->get_env();
 			uint32_t nargs = (uint32_t)env.size();
 
-			for(j = 0; j < nargs; j++)
+			if(nargs == 0)
 			{
-				m_tstr += env[j];
-				if(j < nargs -1)
+				RETURN_EXTRACT_STRING(m_tstr);
+			}
+
+			if(!m_argname.empty()) // contains ENV_NAME we are looking to extract
+			{
+				// proc.env[ENV_NAME] use case: returns matched env variable value
+				for (const auto& item : env)
 				{
-					m_tstr += ' ';
+					std::string str(item.data(), item.data() + item.size());
+					if(str.find(m_argname) == 0)
+					{
+						size_t pos = str.find('=');
```

Comment:
uhm not sure this will work in all the cases, for example, if the env variables are `ENV_VAR_1=dummy_1 ENV_VAR_2=dummy_2 ENV_VAR=dummy` a filter check like `proc.env[ENV_VAR]` will extract `dummy_1` instead of `dummy`, maybe we can add a test for it

Reply:
What you suggested re having a cleaner map instead would seem more robust, maybe let's do that instead? And yes will update tests.

Reply:
for sure a map will simplify the whole implementation, no need to search inside strings, at least i hope :crossed_fingers: